explanation/intro-to/ebpf.md

```diff
@@ -0,0 +1,122 @@
+(introduction-to-ebpf)=
+# Introduction to eBPF
+
+eBPF is a lightweight virtual machine within the kernel. This virtual machine
+is exposed to the user-space and is able to safely run sandboxed programs in
+the kernel-space.
+
+BPF was originally designed to perform network packet filtering. Thus its name:
+_Berkley Packet Filter_. Over time, BPF was extended to use more registers,
+support 64-bit registers, data stores (Maps), and more. Such enhancement
+allowed this extended version of BPF (eBPF) to be decoupled from the kernel
+networking subsystem - eBPF was transformed in a tool with a broader scope,
+capable of enhancing not only the networking experience, but also tracing,
+profiling, observability, security, etc. Given its broader range of
+applications, calling eBPF a packet filter no longer describes its full
+capabilities. eBPF is now considered to be a standalone term and no longer
+refers to _extended Berkley Packet Filter_, reflecting its expanded role. Also,
+the terms eBPF and BPF are used interchangeably, to refer to eBPF.
+
```

Comment:
I'd maybe put a subheading in here like "how it works" or something, just to break up all the text

Reply:
Done. I added more sections after it to reflect what topic is being discussed.